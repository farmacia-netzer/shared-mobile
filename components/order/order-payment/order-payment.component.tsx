import React, { useCallback } from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { transformAmount } from '##common/index';
import { LARGE_MARGIN, MEDIUM_SPACING, NORMAL_MARGIN, NORMAL_PADDING } from '##theme/dimensions.constant';
import { FONT_SIZE } from '##theme/typography.constant';
import { Payment, PaymentStatus, TOrderSummary } from '##types/order.type';
import { NetzerCheckbox } from '##component/netzer-checkbox';
import { updatePayment } from '##redux/slices/order.slice';
import { useDispatch } from '##redux/store';
import { ScrollView } from 'react-native-gesture-handler';

interface OrderPaymentProp {
  orderId: string;
  payments: Payment[];
  summary: TOrderSummary;
}

const Pay = ({ orderId, id, amount, method, checked }) => {
  const dispatch = useDispatch();

  const onClick = useCallback(() => {
    if (checked) return;
    dispatch(updatePayment(orderId, id));
  }, [dispatch, checked, orderId, id]);

  return (
    <NetzerCheckbox
      readOnly={checked}
      label={`${method} ${transformAmount(amount)}`}
      value={checked}
      onChange={onClick}
      relyExternalState={true}
      labelContainerStyle={styles.labelContainerStyle}
    />
  );
};

export const OrderPayment: React.FC<OrderPaymentProp> = ({
  orderId,
  payments,
  summary: { total }
}: OrderPaymentProp) => {
  return (
    <View style={styles.container}>
      <View style={styles.paymentHeader}>
        <Text style={styles.total}>{transformAmount(total)}</Text>
        <Text style={styles.totalSubText}>Total a pagar</Text>
      </View>

      <View style={styles.paymentSection}>
        <Text style={styles.paymentTitle}>Formas de pago</Text>

        <ScrollView>
          {payments.map(({ id, amount, method, status, statusDescription }: Payment) => (
            <View key={id} style={styles.paymentRow}>
              <Pay
                orderId={orderId}
                id={id}
                amount={amount}
                checked={status === PaymentStatus.SUCCESS || status === PaymentStatus.PROCESSED}
                method={method}
              />
              <Text style={styles.statusDescription}>{statusDescription}</Text>
            </View>
          ))}
        </ScrollView>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    width: '100%'
  },
  paymentHeader: {
    marginTop: 150
  },
  total: {
    color: '#4961DC',
    fontSize: FONT_SIZE.XXLARGE + FONT_SIZE.XLARGE,
    textAlign: 'center'
  },
  totalSubText: {
    color: '#4961DC',
    textAlign: 'center',
    fontSize: FONT_SIZE.LARGE
  },
  paymentTitle: {
    fontSize: FONT_SIZE.MEDIUM,
    marginBottom: NORMAL_MARGIN,
    fontWeight: '400',
    lineHeight: 20
  },
  payButton: {
    padding: NORMAL_PADDING,
    backgroundColor: 'red'
  },
  paymentSection: {
    marginTop: MEDIUM_SPACING
  },
  paymentRow: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  text: {
    marginLeft: LARGE_MARGIN,
    fontSize: FONT_SIZE.LARGE
  },
  labelContainerStyle: {
    marginTop: 3
  },
  statusDescription: {
    marginTop: NORMAL_MARGIN + 5
  }
});
