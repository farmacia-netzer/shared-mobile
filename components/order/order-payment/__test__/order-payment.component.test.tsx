import React from 'react';
import { render } from '@testing-library/react-native';

import { OrderPayment } from '../order-payment.component';

describe('<OrderPayment />', () => {
  const payments = [];
  const orderId = 'UUID';
  const summary: any = { total: 'RDS 10' };

  it('renders', () => {
    const { toJSON } = render(<OrderPayment payments={payments} orderId={orderId} summary={summary} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
