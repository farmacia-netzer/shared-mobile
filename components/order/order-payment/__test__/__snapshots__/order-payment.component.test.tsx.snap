// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderPayment /> renders 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#4961DC",
          "fontSize": 52,
          "textAlign": "center",
        }
      }
    >
      S10
    </Text>
    <Text
      style={
        Object {
          "color": "#4961DC",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Total a pagar
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 20,
          "marginBottom": 10,
        }
      }
    >
      Formas de pago
    </Text>
    <RCTScrollView
      collapsable={false}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      waitFor={
        Array [
          Object {
            "current": null,
          },
        ]
      }
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
